lib/file_store/s3_store.rb

```diff
@@ -35,17 +36,41 @@ def store_upload(file, upload, content_type = nil)
       url
     end
 
+    def move_existing_stored_upload(existing_external_upload_key, upload, content_type = nil)
+      upload.url = nil
+      path = get_path_for_upload(upload)
+      url, upload.etag = store_file(
```

Comment:
I think we should avoid methods that mutate state of objects. If we need to mutate the `upload` object, I think the method name should make it obvious that it is doing so.

Reply:
Ah I did that because the existing store_upload method does it too:

https://github.com/discourse/discourse/blob/0da1e94bdde85caacab9fea71a969dae9b1e3c56/lib/file_store/s3_store.rb#L25-L37

Reply:
ahh ok lets keep it as it is then.